subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImplementation.kt

```diff
@@ -21,47 +22,50 @@ class TestRunnerImplementation(
     private val logger = loggerFactory.create<TestRunner>()
 
     override suspend fun runTests(tests: List<TestRunRequest>, scope: CoroutineScope): TestRunnerResult {
-
         logger.debug("started")
+        return withContext(scope.coroutineContext) {
```

Comment:
Run in the scope's context for coupling runner execution

Reply:
how about return deffered? don't you think it will be better?

Reply:
I wrote it as a deferred but IDEA gave me the advice to change it to this version